### PR TITLE
Don't build structure-aware libpng target when building the coverage engine

### PR DESCRIPTION
**Pull request by adrianherrera**
libpng fails to build when setting `FUZZING_ENGINE=coverage` due to a linking error with missing `LLVMFuzzerMutate` symbol.

To fix this I've added a flag (`BUILD_STANDALONE_TARGET`) that is set in build_coverage and checked before building the structure-aware version of libpng. I chose a flag rather than checking if `FUZZING_ENGINE == coverage` in libpng's build.sh so that I could also set this flag for other custom fuzzing engines that I am testing with your test suite.

### CODE REVIEW

**Comment by Dor1s** [comment]
Hi @adrianherrera,

I've accidentally noticed your message regarding "other custom fuzzing engines that I am testing with your test suite". If you're interested in sharing the results of your experiments and are willing to make the experiments reproducible (e.g. provide a shell script, or a sequence of steps to be done), please consider contributing to https://github.com/google/fuzzing repository. It's our "fuzzing forum" kinda thing, and any publications about fuzzing engines comparison with reproducible results would be highly appreciated there :)

**Comment by adrianherrera** [comment]
Hey @Dor1s,

I definitely plan on sharing the results once I have them. Thanks for the repo link, I wasn't aware of that one!

**Comment by Dor1s** [comment]
@adrianherrera that sounds great! Yeah, that repo is fairly empty as of now, so we haven't advertised it anyhow yet, but we plan on adding more content as well in the meantime.

File: libpng-1.2.56/build.sh [hunk 1]
```diff
@@ -21,5 +21,7 @@ if [[ $FUZZING_ENGINE == "hooks" ]]; then
 fi
 set -x
 $CXX $CXXFLAGS -std=c++11 $SCRIPT_DIR/target.cc BUILD/.libs/libpng12.a $LIB_FUZZING_ENGINE -I BUILD/ -I BUILD -lz -o $EXECUTABLE_NAME_BASE
-$CXX $CXXFLAGS -std=c++11 $SCRIPT_DIR/target.cc BUILD/.libs/libpng12.a $LIB_FUZZING_ENGINE -I BUILD/ -I BUILD -lz -o $EXECUTABLE_NAME_BASE-structure-aware \
-  -include $SCRIPT_DIR/png_mutator.h -DPNG_MUTATOR_DEFINE_LIBFUZZER_CUSTOM_MUTATOR
+if [ -z $BUILD_STANDALONE_TARGET ]; then
```
Comment (morehouse) [inline libpng-1.2.56/build.sh]: Hmm... So with this patch we effectively disable coverage builds for the custom mutator fuzzer.  Would it be better to fix the coverage build instead?

Comment (adrianherrera) [reply libpng-1.2.56/build.sh]: Maybe! Let me look into it and I'll let you know

**Comment by adrianherrera** [comment]
Hey @morehouse,

Sorry about the delay. I've change the approach so that a coverage build goes through and it links with a custom version of `LLVMFuzzerMutate` (which I just copied from compiler-rt's afl_driver.cpp).

Let me know what you think!

**Review by morehouse — APPROVED** [review]
LGTM

**Pull request merged**